python/lsst/meas/algorithms/testUtils.py

```diff
@@ -178,3 +184,116 @@ def makeDefectList():
                   ]
 
     return defectList
+
+
+class MockLoadReferenceObjects(ReferenceObjectLoader):
+    """A simple mock of LoadReferenceObjects for tests.
+
+    This class will read in files and return subsets to mock the behavior
+    of LoadReferenceObjects for tests.
+
+    Parameters
+    ----------
+    filenames : `list` [`str`]
+        Names of files to load.
+    config : `lsst.meas.astrom.LoadReferenceObjectsConfig`
+        Configuration object if necessary to override defaults.
+    convert : `bool`
+        Convert from old refcat format?
+    """
+    def __init__(self, filenames, config=None, convert=False):
+        self._cat = self._loadFiles(filenames, convert=convert)
+        self.config = config
+        if self.config is None:
+            self.config = LoadReferenceObjectsConfig()
+
+    def loadSkyCircle(self, ctrCoord, radius, filterName, epoch=None):
+        with Matcher(ctrCoord.getLongitude().asDegrees(), ctrCoord.getLatitude().asDegrees()) as matcher:
+            idx = matcher.query_radius(
+                np.rad2deg(self._cat['coord_ra']),
+                np.rad2deg(self._cat['coord_dec']),
+                radius.asDegrees()
+            )
+        sel = np.zeros(len(self._cat), dtype=bool)
+        sel[idx[0]] = True
+        cat = self._cat[sel].copy(deep=True)
+
+        return Struct(refCat=cat, fluxField=f'{filterName}_flux')
+
+    def loadRegion(self, region, filterName, filtFunc=None, epoch=None):
+        return Struct(refCat=self._cat, fluxField=f'{filterName}_flux')
```

Comment:
This is also meant to be a SourceCatalog. I'm clearly missing something important.